cclib/method/volume.py

```diff
@@ -361,3 +365,57 @@ def electrondensity(ccdata, volume, mocoeffslist):
         edens = electrondensity_spin(ccdata, volume, [mocoeffslist[0]])
         edens.data *= 2
         return edens
+
+
+def read_from_cube(filepath):
+    """Read data from cube files and construct volume object
+    
+       Specification of cube file format is described in http://paulbourke.net/dataformats/cube/
+       
+    Input:
+        filepath -- path to the cube file
+        
+    Output:
+        vol -- Volume object filled with data from cube file
+    """
+    
+    # Cube files are written in bohrs
+    def convert(x):
+        return convertor(x, "bohr", "Angstrom")
+
+    with open(filepath) as f:
+        lines = f.readlines()
+        
+        # First two lines are comments
+        # Lines 3-6 specify the grid in Cartesian coordinates
+        # Line 3 -- [Number of atoms] [Origin x] [Origin y] [Origin z]
+        natom = (int) (lines[2].split()[0])
+        originx, originy, originz = numpy.asanyarray(lines[2].split()[1:], dtype = float)
+        
+        # Line 4, 5, 6 -- [Number of Grid Points] [Spacing x] [Spacing y], [Spacing z]
+        ngridx, spacingx = numpy.asanyarray(lines[3].split(), dtype = float)[[0, 1]]
+        ngridy, spacingy = numpy.asanyarray(lines[4].split(), dtype = float)[[0, 2]]
+        ngridz, spacingz = numpy.asanyarray(lines[5].split(), dtype = float)[[0, 3]]
+        
+        # Line 7, 8, 9 -- Optional Information not used yet by cclib Volume
+        # [Atomic number] [Charge] [Position x] [Position y] [Position z]
+        skiplines = 6
+        while (len(lines[skiplines].split()) == 5):
+            skiplines += 1
+        
+        # Line 10+ (or 7+ if atomic coordinates data are not present)
+        tmp = []
+        for line in lines[skiplines:]:
+            tmp.extend(line.split())
+        tmp = numpy.asanyarray(tmp, dtype = float)
+    
+    # Initialize volume object
+    vol = Volume((convert(originx - 0.5 * ngridx * spacingx), convert(originy - 0.5 * ngridy * spacingy), convert(originz - 0.5 * ngridz * spacingz)), (convert(originx + 0.5 * ngridx * spacingx - 0.5 * spacingx), convert(originy + 0.5 * ngridy * spacingy - 0.5 * spacingy), convert(originz + 0.5 * ngridz * spacingz - 0.5 * spacingz)), (convert(spacingx), convert(spacingy), convert(spacingz)))

```

Comment:
Can you run Black on this?

Reply:
Yes -- I have just run black for this code. Thanks.